### PR TITLE
docs: add section on peerDependenciesMeta field

### LINKED ISSUE #1247: [FEATURE] Document the peerDependenciesMeta in the package.json
I cant find info about the `peerDependenciesMeta` field in the package.json:

https://docs.npmjs.com/files/package.json

### References

It was mentioned at the npm blog here:

https://blog.npmjs.org/post/186983646370/npm-cli-roadmap-summer-2019

>New features likely to land on the 6.x branch:
>
>Add support for the peerDependenciesMeta field. ([#224](https://github.com/npm/cli/pull/224))

Also here in the changelog:

https://github.com/npm/cli/releases/tag/v6.11.0

Also the release announcement:

https://npm.community/t/release-6-11-0/9572

### PR DESCRIPTION
**Pull request by foxxyz**
## What

This PR adds documentation for the `peerDependenciesMeta` field of `package.json`.

## Why

This field was announced in the [Summer 2019 roadmap](https://blog.npmjs.org/post/186983646370/npm-cli-roadmap-summer-2019) and landed in #224, but does not seem to have been documented. I actually needed this field and thought npm did not support it, because [Yarn documents it](https://classic.yarnpkg.com/en/docs/package-json/#toc-peerdependenciesmeta) and npm does not. However, checking the source made me realize it was definitely implemented, but not documented (#1247).

## References

Fixes #1247

### CODE REVIEW

**Comment by darcyclarke** [comment]
@foxxyz hey! Thanks for this. We'll be pulling this into the v7 beta... no changes required.

**Comment by foxxyz** [comment]
Awesome, thanks for the review and let me know if there's anything you'd like me to add!

File: docs/content/configuring-npm/package-json.md [hunk 1]
```diff
@@ -718,6 +718,30 @@ the host package's major version will break your plugin. Thus, if you've worked
 with every 1.x version of the host package, use `"^1.0"` or `"1.x"` to express
 this. If you depend on features introduced in 1.5.2, use `">= 1.5.2 < 2"`.
 
+### peerDependenciesMeta
+
+When a user installs a your package, npm will emit warnings if packages specified in `peerDependencies` are not already installed. The `peerDependenciesMeta` field serves to provide npm more information on how your peer dependencies are to be used. Specifically, it allows peer dependencies to be marked as optional.
```
Comment (RemyRylan) [inline docs/content/configuring-npm/package-json.md]: Tiny typo, needs to be changed from "When a user installs **a** your package" to "When a user installs your package".

Comment (foxxyz) [reply docs/content/configuring-npm/package-json.md]: Good find, thank you for reviewing! I've fixed it 😄

**Review by ruyadorno — APPROVED** [review]

**Pull request closed without merging**